spec/factories/early_career_teachers/create.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+FactoryBot.define do
+  factory :ect, class: "EarlyCareerTeachers::Create" do
+    user   { create(:user) }
+    mentor { create(:mentor) }
+    school_cohort
+
+    initialize_with do
+      EarlyCareerTeachers::Create.call(
+        email: user.email,
+        full_name: user.full_name,
+        mentor_profile_id: mentor.id,
+        school_cohort:,
+      )
+    end
```

Comment:
Use production code to create realistic test setup and not have to maintain complicated factories that will drift away from production has the time goes not. It's not a silver bullet but it's still more maintainable.